From bb507c8cd08505324b206f645dcf7cf9f03ecc9c Mon Sep 17 00:00:00 2001
From: David Andrey <david.andrey@netmodule.com>
Date: Fri, 7 Dec 2012 15:06:03 +0100
Subject: [PATCH 03/21] zynq: BSP: QSPI Flash and NAND muxing

Added pin setup muxing between QSPI and NAND for the
ZX3_PM3 and Z4E board

Signed-off-by:David Andrey david.andrey@netmodule.com
---
 common/cmd_nand.c               |    4 ++++
 common/cmd_sf.c                 |    4 ++++
 include/configs/zx3_pm3.h       |    2 ++
 include/configs/zynq4ethernet.h |    7 ++++++-
 4 files changed, 16 insertions(+), 1 deletions(-)

diff --git a/common/cmd_nand.c b/common/cmd_nand.c
index e24ed7f..c1cda50 100644
--- a/common/cmd_nand.c
+++ b/common/cmd_nand.c
@@ -454,6 +454,10 @@ int do_nand(cmd_tbl_t * cmdtp, int flag, int argc, char * const argv[])
 
 	cmd = argv[1];
 
+#if defined (CONFIG_ZX3_PM3) || defined (CONFIG_Z4E)
+	zx3_set_storage (ZX3_NAND);
+#endif
+
 	/* Only "dump" is repeatable. */
 	if (repeat && strcmp(cmd, "dump"))
 		return 0;
diff --git a/common/cmd_sf.c b/common/cmd_sf.c
index 5ac1d0c..8ea2cde 100644
--- a/common/cmd_sf.c
+++ b/common/cmd_sf.c
@@ -288,6 +288,10 @@ static int do_spi_flash(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[
 	--argc;
 	++argv;
 
+#if defined (CONFIG_ZX3_PM3) || defined (CONFIG_Z4E)
+	zx3_set_storage (ZX3_QSPI);
+#endif
+
 	if (strcmp(cmd, "probe") == 0) {
 		ret = do_spi_flash_probe(argc, argv);
 		goto done;
diff --git a/include/configs/zx3_pm3.h b/include/configs/zx3_pm3.h
index 9d187af..5cfef08 100644
--- a/include/configs/zx3_pm3.h
+++ b/include/configs/zx3_pm3.h
@@ -19,6 +19,8 @@
 #ifndef __CONFIG_ZYNQ4ETHERNET_H
 #define __CONFIG_ZYNQ4ETHERNET_H
 
+#define CONFIG_ZX3_PM3 /* ZX3 PM3 Board */
+
 #define PHYS_SDRAM_1_SIZE (256 * 1024 * 1024)
 
 #define CONFIG_ZYNQ_SERIAL_UART1
diff --git a/include/configs/zynq4ethernet.h b/include/configs/zynq4ethernet.h
index 96cb9fd..124666a 100644
--- a/include/configs/zynq4ethernet.h
+++ b/include/configs/zynq4ethernet.h
@@ -1,7 +1,8 @@
 /*
  * (C) Copyright 2012 Xilinx
+ * (C) Copyright 2012 NetModule
  *
- * Configuration for Zynq Evaluation and Development Board - ZedBoard
+ * Configuration for NetModule Zynq4Ethernet Board - Z4E
  * See zynq_common.h for Zynq common configs
  *
  * This program is free software; you can redistribute it and/or
@@ -18,6 +19,8 @@
 #ifndef __CONFIG_ZYNQ4ETHERNET_H
 #define __CONFIG_ZYNQ4ETHERNET_H
 
+#define CONFIG_Z4E /* Zynq4Ethernet Board */
+
 #define PHYS_SDRAM_1_SIZE (256 * 1024 * 1024)
 
 #define CONFIG_ZYNQ_SERIAL_UART0
@@ -89,6 +92,8 @@
 
 #define CONFIG_SYS_LONGHELP     /* undef to save memory */
 
+#define CONFIG_ENV_OVERWRITE
+
 #undef CONFIG_SYS_MAXARGS
 #define CONFIG_SYS_MAXARGS 32
 
-- 
1.7.4.1

